### PR TITLE
SI-8582 emit InnerClasses attribute in GenBCode

### PR DESCRIPTION
**Pull request by lrytz**
It seems it was simply forgotten, the `InnerClasses` attribute is emitted for mirror classes and bean info classes.

I removed the `-bcode` test since we have a build that passes `-Ybackend:GenBCode` to all tests.

Review by @retronym.

### CODE REVIEW

**Comment by lrytz** [comment]
I checked that `sbt test` works on slick using [your instructions](https://github.com/retronym/slick/commit/c9bf4afa8c92cd9e4cc5b2b0ea38d9463563721c),

File: src/compiler/scala/tools/nsc/backend/jvm/BytecodeWriters.scala [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* NSC -- new Scala compiler
  * Copyright 2005-2013 LAMP/EPFL
- * @author  Paul Phillips
+ * @author  Martin Odersky
```
Comment (soc) [inline src/compiler/scala/tools/nsc/backend/jvm/BytecodeWriters.scala:3]: ???

Comment (lrytz) [inline src/compiler/scala/tools/nsc/backend/jvm/BytecodeWriters.scala:3]: This is the generic header of each file in the compiler, it should be the same across the compiler.

There are specific `@author` tags in the comments of individual classes.

**Comment by lrytz** [comment]
Updated. Seems github lost track of our previous discussions:
- https://github.com/lrytz/scala/commit/cb53f1ca7de099d7ae82670740dd59528a86f47a#diff-4e05b68598824052b89050d8d914700aR22
- https://github.com/lrytz/scala/commit/3832dc62cbb5bfb79269021f2e8e4d01f8541a3c#diff-f1280415c1d5a54238d1cab8d4d8fde2R17 ff

**Comment by retronym** [comment]
LGTM, nice "above and beyond" in poking around at the `BeanInfo` implementation to find the hole.

**Pull request merged**